src/library/Utilities/RemoteContainerConnectionDetailsResolver.cs

```diff
@@ -293,6 +293,15 @@ internal class RemoteContainerConnectionDetailsResolver
                         _log.Warning("{0} pods found, but they are all running RemoteAgent", pods.Count);
                         throw new UserVisibleException(_operationContext, Resources.SpecifiedServiceBackedByMultipleRemoteAgents, service.Metadata?.Name, pods.Count);
                     }
+
+                    // filter pods which are not replicaset or statefulset
+                    podsNotRunningDevHostAgent = podsNotRunningDevHostAgent.Where(p => p.IsOwnerOfKind("ReplicaSet", "StatefulSet"));
+                    if (podsNotRunningDevHostAgent.Count() == 0)
+                    {
+                        _log.Warning("{0} pods found, after filtering pods which are of kind replicaset or statefulset", pods.Count);

```

Comment:
maybe simplify the message -- {0} pods found of kind replicaset or statefulset

Reply:
updated the message. Thanks for the suggestion.